src/test.js

```diff
@@ -692,6 +696,49 @@ export function test( testName, callback ) {
 	newTest.queue();
 }
 
+function skip( testName, data ) {
+	if ( focused || config.currentModule.ignored ) {
+		return;
+	}
+
+	const test = new Test( {
+		testName: testName,
+		skip: true,
+		params: data
+	} );
+
+	test.queue();
+}
+function only( testName, data, callback ) {
+	if ( config.currentModule.ignored ) {
+		return;
+	}
+	if ( !focused ) {
+		config.queue.length = 0;
+		focused = true;
+	}
+
+	const newTest = new Test( {
+		testName: testName,
+		callback: callback,
+		params: data
+	} );
+
+	newTest.queue();
+}
+
+function normalizeTable( data ) {
+	const isTable = data.every( Array.isArray );
+	if ( isTable ) {
+		return data;
+	}
+	return data.map( datum => [ datum ] );
+}
```

Comment:
Do we need some guards to validate that the user-specified `data` is an Array before we start iterating at all? Thinking of edge cases...

```js
QUnit.test.each('raw object', {foo: 2}, assert => { ... })

QUnit.test.each('string', 'hello world', assert => { ... })
```

Otherwise they'd see ".every is not a function" instead of a tailored message, like "Data provided to QUnit.test.each must be an array."

Reply:
That makes sense. I'm `throw`ing an error now, let me know if that is not the preferred way to signal a problem with the test. Please also take a look at the test and let me know if it looks like a good way to test this behavior.